base/src/main/java/org/aya/distill/CoreDistiller.java

```diff
@@ -151,7 +152,9 @@ yield visitCalls(false, term(Outer.AppHead, head), args.view(), outer,
       }
       case CallTerm.Struct structCall -> visitArgsCalls(structCall.ref(), STRUCT_CALL, structCall.args(), outer);
       case CallTerm.Data dataCall -> visitArgsCalls(dataCall.ref(), DATA_CALL, dataCall.args(), outer);
-      case LitTerm.ShapedInt shaped -> shaped.with(
+      case LitTerm.ShapedInt shaped -> options.map.get(DistillerOptions.Key.ShowLiterals)
+        ? Doc.plain(String.valueOf(shaped.repr()))
+        : shaped.with(
         (zero, suc) -> shaped.repr() == 0
           ? linkLit(0, zero.ref, CON_CALL)
           : linkLit(shaped.repr(), suc.ref, CON_CALL),
```

Comment:
Why change this? The literals won't be linked to its constructor when `ShowLiteral` is set.

Reply:
Like this 
![image](https://user-images.githubusercontent.com/9902643/168727264-7b582c29-9599-46d7-b161-679e96a1f059.png)